src/overlays/actors/ovl_En_Okarina_Tag/z_en_okarina_tag.c

```diff
@@ -30,14 +29,123 @@ const ActorInit En_Okarina_Tag_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void EnOkarinaTag_Destroy(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Okarina_Tag/EnOkarinaTag_Destroy.s")
+void EnOkarinaTag_Init(Actor* thisx, PlayState* play) {
+    EnOkarinaTag* this = THIS;
+    f32 zRot;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Okarina_Tag/EnOkarinaTag_Init.s")
+    zRot = 0.0f;
+    i = 0;
+    this->actor.flags &= ~1;
+    this->unk148 = ENOKARINATAG_GET_F800(thisx);
+    this->unk14A = ENOKARINATAG_GET_780(thisx);
+    this->unk14C = ENOKARINATAG_GET_7F(thisx);
+    if (this->actor.world.rot.z > 0) {
+        zRot = this->actor.world.rot.z;
+        while (zRot > 10.0f) {
+            zRot -= 10.0f;
+            i++;
+        }
+    }
+    this->unk154 = zRot * 50.0f;
+    this->unk158 = i * 50.0f;
+    if (this->unk14C == 0x7F) {
+        this->unk14C = -1;
+    }
+    if (this->unk14A == 0xF) {
+        this->unk14A = -1;
+    }
+    this->actor.targetMode = 1;
+    this->actionFunc = func_8093E518;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Okarina_Tag/func_8093E518.s")
+void func_8093E518(EnOkarinaTag* this, PlayState* play) {
+    f32 xzRange;
+    f32 yRange;
+    s16 yDiff;
+    u16 var_v1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Okarina_Tag/func_8093E68C.s")
+    if (this->unk14C >= 0) {
+        if (this->unk148 == 0) {
+            if (Flags_GetSwitch(play, this->unk14C) != 0) {
```

Comment:
Not requiring it, but we know the params value `ENOKARINATAG_GET_7F` is a switch flag because it gets passed as a parameter to `Flags_GetSwitch` or `Flags_SetSwitch`. Where a switch flag is a flag in the scene that keeps track of changes made by the player, so when they leave the area and come back the changes they made to the scene like opening a gate or triggering a chest to appear aren't undone by the scene being destroyed and rebuilt.

You can name this->unk14C to `this->switchFlags` and ENOKARINATAG_GET_7F to `ENOKARINATAG_GET_SWITCHFLAGS`. The other params in this actor are less obvious, but switch flags/ paths are fairly obvious because they always pass their value to an already discovered/named function